File: Cpp/001~100/001. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> my_map;
        for (int i = 0; i < nums.size(); i++) { 
            if (my_map.count(target - nums[i]) && i != my_map[target - nums[i]]) {
                 return {i, my_map[target - nums[i]]};
            }
            my_map[nums[i]] = i;
        }
    }
};

File: Python/001~050/001. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i, v in enumerate(nums): 
            if v in m:
                return [m[v], i]
            m[target-v] = i
        return -1, -1
		
'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(num)):
            if num[i] in m: return m[num[i]], i
            m[target - num[i]] = i  
        return -1, -1
'''